og.module

```diff
@@ -142,8 +142,13 @@ function og_entity_access(EntityInterface $entity, $operation, AccountInterface
 
   $entity_type_id = $entity->getEntityTypeId();
   $bundle_id = $entity->bundle();
+  $is_group_content = Og::isGroupContent($entity_type_id, $bundle_id);
 
-  if (!Og::isGroup($entity_type_id, $bundle_id) && !Og::isGroupContent($entity_type_id, $bundle_id)) {
+  if (!Og::isGroup($entity_type_id, $bundle_id) && !$is_group_content) {
+    return AccessResult::neutral();
+  }
+
+  if (empty(Og::getEntityGroups($entity)) && $is_group_content) {
```

Comment:
If you just need to know whether an entity has _any_ groups then it is better to do this:

```
$group_count = \Drupal::service('og_membership_manager')->getGroupCount($entity);
if ($group_count && !$is_group_content) {}
```

This has a few advantages: it will actually check against the database that there are no stale references left behind by the Entity Reference module causing false positives, and it will avoid loading the full entity objects.